lib/components/ConnectionParamsControl.jsx

```diff
@@ -0,0 +1,242 @@
+import Col from 'react-bootstrap/Col';
+import TextInput from './input/TextInput';
+
+const SUCCESS = 'success';
+const ERROR = 'error';
+
+class ConnectionParamsControl extends React.PureComponent {
+    state = {
+        minConnectionInterval: this.props.connectionParameters
+            .minConnectionInterval,
+        maxConnectionInterval: this.props.connectionParameters
+            .maxConnectionInterval,
+        slaveLatency: this.props.connectionParameters.slaveLatency,
+        connectionSupervisionTimeout: this.props.connectionParameters
+            .connectionSupervisionTimeout,
+    };
+
+    validateMinConInterval() {
+        const { minConnectionInterval, maxConnectionInterval } = this.state;
+        if (
+            minConnectionInterval < 7.5 ||
+            minConnectionInterval > 4000 ||
+            minConnectionInterval > maxConnectionInterval
+        ) {
+            return ERROR;
+        }
+        return SUCCESS;
+    }
+
+    validateMaxConInterval() {
+        const { minConnectionInterval, maxConnectionInterval } = this.state;
+        if (
+            maxConnectionInterval < 7.5 ||
+            maxConnectionInterval > 4000 ||
+            minConnectionInterval > maxConnectionInterval
+        ) {
+            return ERROR;
+        }
+        return SUCCESS;
+    }
+
+    validateConSupervisionTimeout() {
+        // bigger than: (1+latency)*max_con_interval*2, or 100ms
+        // multiple of 10 ms
+        const {
+            connectionSupervisionTimeout,
+            slaveLatency,
+            maxConnectionInterval,
+        } = this.state;
+        const bound = (1 + slaveLatency) * 2 * maxConnectionInterval;
+        const lowerBound = bound > 100 ? bound : 100;
+        if (
+            connectionSupervisionTimeout < lowerBound ||
+            connectionSupervisionTimeout > 32000 ||
+            connectionSupervisionTimeout % 10 !== 0
+        ) {
+            return ERROR;
+        }
+        return SUCCESS;
+    }
+
+    validateSlaveLatency() {
+        // less than: (supervision_timeout/(max_con_interval*2))-1, or 500
+        const {
+            slaveLatency,
+            connectionSupervisionTimeout,
+            maxConnectionInterval,
+        } = this.state;
+        const bound =
+            connectionSupervisionTimeout / (maxConnectionInterval * 2) - 1;
+        const upperbound = bound < 499 ? bound : 499;
+        if (
+            slaveLatency < 0 ||
+            slaveLatency > upperbound ||
+            slaveLatency % 1 !== 0
+        ) {
+            return ERROR;
+        }
+        return SUCCESS;
+    }
+
+    handleChange(variableName, value) {
+        this.setState({ [variableName]: +value }, () => {
+            if (
+                this.validateSlaveLatency() === ERROR ||
+                this.validateMinConInterval() === ERROR ||
+                this.validateMaxConInterval() === ERROR ||
+                this.validateConSupervisionTimeout() === ERROR
+            ) {
```

Comment:
[minor] This could perhaps be replaced by this:

```js
if (
    [
        this.validateSlaveLatency(),
        this.validateMinConInterval(),
        this.validateMaxConInterval(),
        this.validateConSupervisionTimeout(),
    ].includes(ERROR)
) {
```

Not necessarily more readable, but it reduces some duplication.